compiler/rustc_interface/src/interface.rs

```diff
@@ -219,7 +225,11 @@ pub fn parse_check_cfg(specs: Vec<String>) -> CheckCfg {
             );
         }
 
-        cfg.names_valid.extend(cfg.values_checked.iter().cloned());
+        if let Some(values_valid) = &cfg.values_valid {
+            if let Some(names_valid) = &mut cfg.names_valid {
```

Comment:
Can be merged into a single `if let`.

Reply:
I tried that at first, but it's now yet complete in beta.
```rust
error[E0658]: `let` expressions in this position are experimental
   --> compiler/rustc_interface/src/interface.rs:228:12
    |
228 |         if let Some(values_valid) = &cfg.values_valid && let Some(names_valid) = &mut cfg.names_valid {
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
    = help: add `#![feature(let_chains)]` to the crate attributes to enable
    = help: you can write `matches!(<expr>, <pattern>)` instead of `let <pattern> = <expr>`
```
Then:
```rust
error: the feature `let_chains` is incomplete and may not be safe to use and/or cause compiler crashes
 --> compiler/rustc_interface/src/lib.rs:4:12
  |
4 | #![feature(let_chains)]
  |            ^^^^^^^^^^
  |
  = note: `-D incomplete-features` implied by `-D warnings`
  = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information

```

Reply:
Oh! Addressed in https://github.com/rust-lang/rust/pull/94175/commits/7de5f3b815f0146d2c964e625d2a99d89361aeb5.